crates/evm/src/call.rs

```diff
@@ -166,81 +162,46 @@ impl<C: sov_modules_api::Context> Evm<C> {
             &mut citrea_handler_ext,
             cumulative_gas_used,
             &mut self.blob_gas_used,
-        );
+        )?;
 
         // Iterate each evm_txs_recovered and results pair
         // Create a PendingTransaction for each pair
         // Push each PendingTransaction to pending_transactions
         for (evm_tx_recovered, result) in users_txs.into_iter().zip(results.into_iter()) {
-            match result {
-                Ok(result) => {
-                    // take ownership of result.log() and use into()
-                    let logs: Vec<_> = result.logs().iter().cloned().map(Into::into).collect();
-                    let logs_len = logs.len() as u64;
-
-                    let gas_used = result.gas_used();
-                    cumulative_gas_used += gas_used;
-                    let tx_hash = evm_tx_recovered.hash();
-                    let tx_info = citrea_handler_ext.get_tx_info(tx_hash).unwrap_or_else(|| {
-                        panic!("evm: Could not get associated info for tx: {tx_hash}")
-                    });
-
-                    let receipt = Receipt {
-                        receipt: reth_primitives::Receipt {
-                            tx_type: evm_tx_recovered.tx_type(),
-                            success: result.is_success(),
-                            cumulative_gas_used,
-                            logs,
-                        },
-                        gas_used: gas_used as u128,
-                        log_index_start,
-                        l1_diff_size: tx_info.l1_diff_size,
-                    };
-                    log_index_start += logs_len;
+            // take ownership of result.log() and use into()
+            let logs: Vec<_> = result.logs().iter().cloned().map(Into::into).collect();
+            let logs_len = logs.len() as u64;
 
-                    let pending_transaction = PendingTransaction {
-                        transaction: TransactionSignedAndRecovered {
-                            signer: evm_tx_recovered.signer(),
-                            signed_transaction: evm_tx_recovered.into(),
-                            block_number: block_number.saturating_to(),
-                        },
-                        receipt,
-                    };
+            let gas_used = result.gas_used();
+            cumulative_gas_used += gas_used;
+            let tx_hash = evm_tx_recovered.hash();
+            let tx_info = citrea_handler_ext
+                .get_tx_info(tx_hash)
+                .unwrap_or_else(|| panic!("evm: Could not get associated info for tx: {tx_hash}"));
```

Comment:
```suggestion
                .expect("evm: Could not get associated info for tx: {tx_hash}");
```
But in general, i would rather this
```rust
.unwrap_or(anyhow!("evm: Could not get associated info for tx: {tx_hash}");
```
Since we are now returning errors.

Reply:
so I used errors for things that would invalidate the soft confirmation e.g. a tx that doesn't work cannot be in a soft confirmation. 

however, in this example, we set tx info in the code before execution and then we get it after execution. if it doesn't happen, again it's a very very big problem; it will keep happening with correct soft. conf. as well. propagating the error doesn't help with anything